network/tcp.go

```diff
@@ -100,11 +104,14 @@ func (c *TCPConn) Receive() (nm Packet, e error) {
 func (c *TCPConn) receiveRaw() ([]byte, error) {
 	c.receiveMutex.Lock()
 	defer c.receiveMutex.Unlock()
+	c.conn.SetReadDeadline(time.Now().Add(readTimeout))
```

Comment:
It's a bit *deceiving* because it's **not** a timeout. Each time we read, we have to set up the new deadline in *absolute* time. In fact I'm doing it wrong, I'm supposed to put that deadline just above the line where we read in the for loop. One commit coming along.